server.go

```diff
@@ -344,14 +348,34 @@ func (s *Server) Serve(lis net.Listener) error {
 		}
 		s.mu.Unlock()
 	}()
+
+	var tempDelay time.Duration // how long to sleep on accept failure
+
 	for {
 		rawConn, err := lis.Accept()
 		if err != nil {
+			if ne, ok := err.(net.Error); ok && ne.Temporary() {
+				if tempDelay == 0 {
+					tempDelay = 5 * time.Millisecond
+				} else {
+					tempDelay *= 2
+				}
+				if max := 1 * time.Second; tempDelay > max {
+					tempDelay = max
+				}
+				s.mu.Lock()
+				s.printf("Accept error: %v; retrying in %v", err, tempDelay)
+				s.delayer.Reset(tempDelay)
```

Comment:
I'm not sure if reusing timer will cause any issues here.
I propose making timer a local variable and creating a new one every time.
To interrupt the timer, instead of resetting the timer, we can add a context to server and do a select
```go
select {
  case <- timer.C:
  case <- s.ctx.Done():
}
```